crates/core/c8y_api/src/smartrest/message.rs

```diff
@@ -167,63 +166,62 @@ mod tests {
 
     #[test]
     fn selected_control_chars_remain() {
-        let input = vec![0x00, 0x09, 0x0A, 0x0D]; // NULL, \t, \n, \r
-        let sanitized = sanitize_for_smartrest(input, MAX_PAYLOAD_LIMIT_IN_BYTES);
+        let input = b"\0\t\n\r";
+        let sanitized = sanitize_bytes_for_smartrest(input, MAX_PAYLOAD_LIMIT_IN_BYTES);
         assert_eq!(sanitized, "\t\n\r".to_string());
     }
 
     #[test]
     fn control_chars_are_removed() {
-        let input: Vec<u8> = (0x00..0xff).collect();
-        let sanitized = sanitize_for_smartrest(input, MAX_PAYLOAD_LIMIT_IN_BYTES);
-        let re = Regex::new(r"[^\x20-\x7E\xA0-\xFF\t\n\r]").unwrap();
-        assert!(!re.is_match(&sanitized));
+        let input: Vec<u8> = (0x00..=0x7f).collect();
+        let sanitized = sanitize_bytes_for_smartrest(&input, MAX_PAYLOAD_LIMIT_IN_BYTES);
+        assert_eq!(
+            sanitized,
+            "\t\n\r !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~"
+        );
```

Comment:
I modified the range this test covers and the assertion because the test was originally supplying invalid UTF-8 (so the sanitize function was returning an empty string), and I don't think this was the intention. The Regex matching obfuscated this since the empty string trivially passed the assertion, so I expanded this into a string literal as this guarantees the output is what we want.